pkg/build/nodeimage/internal/kube/source.go

```diff
@@ -27,25 +29,43 @@ import (
 	"sigs.k8s.io/kind/pkg/exec"
 )
 
-// ImportPath is the canonical import path for the kubernetes root package
-// this is used by FindSource
-const ImportPath = "k8s.io/kubernetes"
-
 // FindSource attempts to locate a kubernetes checkout using go's build package
 func FindSource() (root string, err error) {
-	// look up the source the way go build would
-	pkg, err := build.Default.Import(ImportPath, build.Default.GOPATH, build.FindOnly|build.IgnoreVendor)
-	if err == nil && maybeKubeDir(pkg.Dir) {
-		return pkg.Dir, nil
+	// check current working directory first
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to locate kubernetes source, could not current working directory: %w", err)
+	}
+	if probablyKubeDir(wd) {
+		return wd, nil
+	}
+	// then look under GOPATH
+	gopath := build.Default.GOPATH
+	if gopath == "" {
+		return "", errors.New("could not find Kubernetes source under current working directory and GOPATH is not set")
 	}
-	return "", fmt.Errorf("could not find %s module source under GOPATH=%s: %w", ImportPath, build.Default.GOPATH, err)
+	// try k8s.io/kubernetes first (old canonical GOPATH locaation)
+	if dir := filepath.Join(gopath, "src", "k8s.io", "kubernetes"); probablyKubeDir(dir) {
+		return dir, nil
+	}
+	// then try github.com/kubernetes/kubernetes (CI without path_alias set)
+	if dir := filepath.Join(gopath, "src", "github.com", "kubernetes", "kubernetes"); probablyKubeDir(dir) {
```

Comment:
Observed that some CI jobs are missing `path_alias` and would previously fail to locate sources.

We fixed those by passing the source path to `kind build node-image`, but ideally it should just work.

Prow will default to a GOPATH based on the repo URL if `path_alias` is not specified.